#52: removed scss functions from css styles (moved to variables)

bugy opened the pull request:
As discussed in #52 I tried to remove scss functions from css styles
There are several changes, which I suggest (the PR is draft)

### 1. Generic colors
I also tried to reduce the color variants, so I introduced generic variables:
```
$bg-color
$bg-hover-color-opaque // 4% gray overlay
$bg-hover-color-solid // 4% gray overlay
$bg-focus-color-opaque // 12% gray overlay
$bg-focus-color-solid // 12% gray overlay

$hover-on-secondary-color // e.g. hover on raised button, it should make it lighter
$focus-on-secondary-color // e.g. focus on raised button, it should make it even more lighter
$secondary-color-hover-opaque // 8% overlay, based on secondary color (e.g. time picker hover)
$secondary-color-focus-opaque // 24% overlay, based on secondary color (e.g. time picker focus)
```
I got these ideas from material design docs: https://material.io/components
Maybe, some names should be adjusted for transparency

### 2. Component specific variables
At the moment, all the variables are defined in the global file
As for me, it would be better to move component variables to the corresponding component files, so it's easier to track where they are used. Also, if some people are using only some components, they are not required to override all the variables

### 3. Better hover/focus states
For the changed components, hover/focus state didn't correspond material guidelines (either they didn't exist at all, like hover in a switch component) or they didn't match the spec (e.g. button hover was lighter, but focus was darker).
So I tried to adjust them to better match specs and also re-use Generic colors

I prepared a test-bed with changed components: https://script-server.net/material-test/index.html (please don't be confused by the server name, I have only 1 running web-server, so just reused it for test-bed)

Would be glad to hear any critique, suggestions. Also, would be nice to agree on the next steps. I guess that refactoring every component in such a manner would be too much work

Pierstoval commented on sass/components/_dropdown.scss:
```diff
@@ -1,3 +1,10 @@
+$dropdown-color: $secondary-color !default;
+$dropdown-item-height: 50px !default;
+
+
```
Unnecessary line feed here
```suggestion
```

Pierstoval reviewed (commented):
I'm not sure moving dropdown or tabs variables to their own component is good for now, it breaks BC a bit too much IMO.

I think it's fine to have all variables at the same place for now.

As there was no release in the main Materialize repo, I think breaking BC is not really useful if we want people to migrate to the new fork, so we should keep things as they originally were as much as possible, and change the architecture for another major version.

Also, on your test server, do you think you can add the main materialize theme so we can compare with the original one? For instance, when clicking the button, I don't see this light circle animation.

For the rest, it seems okay to me.

bugy commented:
Hi @Pierstoval thanks for your comments!
How can i apply the main theme? I was comparing components to the materializecss.com demo

bugy commented:
Ah, wait, you mean waves effect? Will check, why it doesn't work

bugy commented:
Hi @Pierstoval I fixed the waves-effect.

The problem was, that I'm importing local file `dist/js/materialize.js`, but this file wasn't updated in the repository for a long time already. Not sure what is the expected flow for this file. I just rebuilt it locally and it worked

I also moved variables back to the central file

Pierstoval reviewed (approved):
According to what I tested on your test server, it looks good to me 👍 

@materializecss/members ?

roiLeo commented on sass/components/forms/_switches.scss:
```diff
@@ -56,36 +56,40 @@
 .switch input[type=checkbox][disabled] + .lever {
   cursor: default;
-  background-color: rgba(0,0,0,.12);
+  opacity: 0.5;
```
has this been test on ie11 ?

bugy replied:
@roiLeo good point!
I checked it and it looks good 
![IMG_20201212_135238_564](https://user-images.githubusercontent.com/1275813/101984364-4e7a0780-3c81-11eb-8d07-71f9d7eca8a3.jpg)

bugy commented:
Hi, can we merge this?
@materializecss/members

Also, could anyone advise on this problem?
> The problem was, that I'm importing local file dist/js/materialize.js, but this file wasn't updated in the repository for a long time already. Not sure what is the expected flow for this file. I just rebuilt it locally and it worked

roiLeo commented:
> Also, could anyone advise on this problem?
> 
> > The problem was, that I'm importing local file dist/js/materialize.js, but this file wasn't updated in the repository for a long time already. Not sure what is the expected flow for this file. I just rebuilt it locally and it worked

Is this  #6622 related ?

bugy commented:
@materializecss/members

Sorry to disturb you, but are there any updates on this issue?

DanielRuf commented:
> Sorry to disturb you, but are there any updates on this issue?

No updates so far as you can see.

DanielRuf commented:
> Is this [Dogfalo#6622](https://github.com/Dogfalo/materialize/pull/6622) related ?

bugy commented:
Hi @DanielRuf, thanks for the response. Sorry, I missed the question. I think Dogfalo#6622 might be related to my problem with the dist, I don't know. I can live with it, just curious.
This is not related to my PR though.

bugy commented:
I screwed up commit history with a rebase. Creating a new PR, with a different branch name instead
